generate-client.sh

```diff
@@ -47,10 +54,23 @@ else
   done
 fi
 
+# Step 4: Build client package
 echo "Creating package.json from template..."
 cd /client
 DATE=$(date -u +"%Y%m%d%H%M%S")
-cat package.json.template | jq --arg pn "$INPUT_NAME" '.name="@birdiecare/\($pn)-client"' | jq --arg pn "$INPUT_NAME" '.description="Autogenerated API client for \($pn) service."' | jq --arg pn "$INPUT_NAME" '.repository.url="ssh://git@github.com/birdiecare/\($pn).git"' | jq --arg date "$DATE" '.version="0.0.\($date)"' > package.json
+cat package.json.template \
+	| jq --arg SERVICE "$INPUT_SERVICE_NAME" \
+	     --arg REGISTRY "$INPUT_REGISTRY_NAMESPACE" \
+	     '.name="@\($REGISTRY)/\($SERVICE)-client"' \
+	| jq --arg SERVICE "$INPUT_SERVICE_NAME" \
+	     '.description="Autogenerated API client for \($SERVICE) service."' \
+	| jq --arg SERVICE "$INPUT_SERVICE_NAME" \
+	     --arg REGISTRY "$INPUT_REGISTRY_NAMESPACE" \
+	     '.repository.url="ssh://git@github.com/\($REGISTRY)/\($SERVICE).git"' \
```

Comment:
That's a strong assumption. Do you think we can make it configurable, with a default value?

Reply:
This is perhaps a bit non-sensical in general. The `repository` property on a package.json file is supposed to specify the url of the repo in case people want to contribute. But we are just creating and publishing an auto-generated client as an NPM package; there is no GitHub repo where you can contribute to the client. 

It's probably best to just leave this property off the package.json. WDYT?

Reply:
Agreed 👍

Reply:
Actually @sroze and @djbeaumont it appears that this property is needed in order to be able to publish to Github Packages. And, in addition, Github Packages only supports packages with a scope. See the documentation [here](https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-npm-for-use-with-github-packages#publishing-a-package).

With that in mind I'd suggest limiting the functionality of this action to publishing to a Github package registry (rather than making it generically able to publish to any repository), as doing so seems to be quite a pain and publishing to GH is probably the biggest use case. Would you agree to that approach?